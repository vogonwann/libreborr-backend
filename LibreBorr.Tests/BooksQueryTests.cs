using HotChocolate;
using HotChocolate.Execution;
using LibreBorr.Web.GraphQl.Mutations;
using LibreBorr.Web.GraphQl.Queries;
using LibreBorr.Web.GraphQl.Subscriptions;
using Microsoft.Extensions.DependencyInjection;
using Snapshooter.Xunit;

namespace LibreBorr.Tests;

public class BooksQueryTests
{
    [Fact]
    public async Task SchemaChangeTest()
    {
        var schema = await new ServiceCollection()
            .AddGraphQL()
            .AddQueryType(d => d.Name("Query"))
                .AddType<BookQuery>()
                .AddType<FriendsQuery>()
            .AddMutationType(d => d.Name("Mutation"))
                .AddType<BookMutation>()
                .AddType<FriendMutation>()
            .AddSubscriptionType<BookSubscription>()
            .AddInMemorySubscriptions()
            .BuildSchemaAsync();
        
        schema.ToString().MatchSnapshot();
    }
    
    [Fact]
    public async Task GetBooksTest()
    {
        var schema = await new ServiceCollection()
            .AddGraphQL()
            .AddQueryType(d => d.Name("Query"))
            .AddType<BookQuery>()
            .AddType<FriendsQuery>()
            .AddMutationType(d => d.Name("Mutation"))
            .AddType<BookMutation>()
            .AddType<FriendMutation>()
            .AddSubscriptionType<BookSubscription>()
            .AddInMemorySubscriptions()
            .ExecuteRequestAsync(@"
                query books {
                  books {
                    id
                    title
                    description
                    tags {
                      name
                    }
                    image
                    authors
                    genres
                  }
                }
                ");
        
        schema.ToJson().MatchSnapshot();
    }
}