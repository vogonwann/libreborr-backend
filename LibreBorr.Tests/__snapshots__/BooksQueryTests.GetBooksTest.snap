{
  "errors": [
    {
      "message": "Unexpected Execution Error",
      "locations": [
        {
          "line": 3,
          "column": 19
        }
      ],
      "path": [
        "books"
      ]
    }
  ],
  "data": {
    "books": null
  }
}
